src/matrixClass.ts

```diff
@@ -0,0 +1,317 @@
+import {Position} from './Position'
+
+export class Matrix{
+    public board:number[][] = [];
+    /*
+    0 = uncharted
+    1 = hit
+    2 = miss
+    3 = suggested target
+    4 = blocked out no ship area
+    */
+    public ships:boolean[] = [true,true,true,true,true];
+    /*
+    index = length
+    0 = 2
+    1 = 3
+    2 = 3
+    3 = 4
+    4 = 5
+     */
+
+
+    
+    public returnUnsunkShot(){
+        for(let row =0; row < this.board.length; row ++){
+            for(let col = 0; col < this.board.length; col ++){
+                if(this.board[row][col] == 3)return this.positionFromCoords(row, col);
+            }
+        }
+        // should never reach hear;
+        console.log("never logged");
+    }
+
+    public surroundHorizontalShips(){
+        let hitCount;
+        for(let row =0; row < this.board.length; row ++){
+            hitCount = 0;
+            for(let col = 0; col < this.board.length; col ++){
+                if(this.board[row][col] == 1)hitCount ++;
+                else if(this.board[row][col] != 1){
+                    if(hitCount > 1)this.horizontalSurroundShip(row,col,hitCount);
+                    
+                    if(hitCount == 2){
+                        this.ships[0] = false;
+                    }
+                    else if(hitCount == 3){
+                        if(this.ships[1])this.ships[1] = false;
+                        else this.ships[2] = false;
+                    }
+                    else if(hitCount == 4) this.ships[3] = false;
+                    else if(hitCount == 5) this.ships[4] = false;
+
+
+
+                    hitCount = 0 ;
```

Comment:
You're putting in a lot of whitespace - I don't think there's much benefit in using more than one line (at tops two, and then only between conceptual boundaries - e.g. class variables & class methods).